### ISSUE TITLE
Clearer communication around allowing non-standard kernel configurations such as PREEMPT_RT

### ISSUE THREAD

**Issue by KyunLFA** [lkrg-org/lkrg#277]
It may become an issue in the future as sort of was previously ( #40 ) if there is no clear wording on an official and readily accessible medium documenting what LKRG allows and doesn't allow in terms of kernel configurations.

Kernel ricers, modders, hackers, researchers etc. may want to use LKRG in ways it is currently not supported, specially custom kernels compiled with aggressive optimizations (one such example where that is normal practice is the entirety of Gentoo Linux), and real-time applications using the PREEMPT_RT patchset, or custom schedulers such as the ones found in CachyOS Linux.

Please document somewhere as time allows what the constraints and burdens to allowing for LKRG to build in custom kernels are and the additional burdens for it to function correctly in different kernel configurations such as PREEMPT_RT, development kernel (RCs), and custom compilation optimizations.

(As one might expect, I have an interest in testing these kinds of configurations, and would be happy to report bugs if LKRG-org rethinks their decision).

**Comment by solardiz** [comment]
In short:

> PREEMPT_RT

Currently unsupported, LKRG will refuse to build.

> development kernel (RCs)

They generally just work, but sometimes recent/development kernels (not only RC) require us to make updates to LKRG as well and in those cases you have to use latest LKRG from this repo rather than our latest release.

> and custom compilation optimizations.

These make it more likely that LKRG loading will fail (unresolved symbols) or that false positives will occur at runtime. It's also possible that protection will be weakened and that will go undetected, but LKRG is best-effort protection anyway.

**Comment by KyunLFA** [comment]
I am aware, but my point is that this information should be in the front page, or the website, somewhere other than the Issues page, but of course that's just my suggestion.

**Comment by solardiz** [comment]
Yes, I get your point, which is why I kept this issue open. I'm not sure what we want to actually do about it.

For recent kernels requiring git LKRG, we'll likely be updating the homepage to state so precisely when this is known to be the case and referring to specific kernel versions. I did such web page edits in the past a few times.

**Comment by osevan** [comment]
Im replying here

https://github.com/lkrg-org/lkrg/issues/40#issuecomment-2396516907

**Comment by osevan** [comment]
6.12 kernel rt mainlined!
